ranzen/torch/data.py

```diff
@@ -165,12 +173,20 @@ def prop_random_split(
     if (sum_ > 1.0) or any(prop < 0 for prop in props):
         raise ValueError("Values for 'props` must be positive and sum to 1 or less.")
     section_sizes = [round(prop * len_) for prop in props]
-    if sum_ < 1:
-        section_sizes.append(len_ - sum(section_sizes))
-    generator = torch.default_generator if seed is None else torch.Generator().manual_seed(seed)
-    indices = torch.randperm(sum(section_sizes), generator=generator).tolist()
+    if (current_len := sum(section_sizes)) < len_:
+        section_sizes.append(len_ - current_len)
+
+    if reproducible:
+        # MT19937 isn't the best random number generator, but it's reproducible, so we're using it.
+        generator = np.random.Generator(np.random.MT19937(seed))
+        indices = np.arange(sum(section_sizes))
+        generator.shuffle(indices)  # Shuffle the indices in-place.
+    else:
```

Comment:
We could just have it always shuffle with nunpy and eliminate this else branch; the only thing that would need to be conditional with that change would be the generator (and could be made so with a simple ternary statement).

Reply:
That's a good idea 👍